### DISCUSSION TITLE
Regarding course work selections

### DISCUSSION THREAD

**Discussion by sab1a** [snigi-gupta/AttendingUniversityAtBuffalo#52] [Q&A] [answered]
Hello, for my Fall 2021 semester I was planning to select the below courses:

1. CSE 531 Analysis of Algorithms (Theory core)
2. CSE 573 Introduction to Computer Vision and Image Processing (Artificial Intelligence Area (AI) core)
3. CSE 510 LEC Software Security  - 1 Elective
4. CSE 586 Distributed Systems - (Hardware and Networked Systems Area (HW) core)

Also, I see the review for CVIP is difficult, but other courses within AI core such as ML overlap with the elective I want to take this semester. Do you think this course load would be managable for the first sem?

Thank you

**Comment by snigi-gupta** [comment, marked as answer]
Hi @sab1a

This seems like a hard-medium studyload. If this is your 1st sem then I will suggest to go with medium difficulty level. 

Comp-sec, DS and Algos are all hard considering you are not familiar with most of the topics that will be taught here. 

And then having CVIP is a medium workload. But it gets slightly challenging if you take CVIP by Doreman. If you take CVIP by Yuan then exams will bhi slightly challenging. So you choose. Both profs are great profs. 

I would suggest you to swap one of the subjects to an easy course to make the workload medium. You can also choose a Seminar if you feel like, but be careful of the limitations it has for not contributing to your gpa. 

I hope this helps. I'd request you to follow me on @yourtechdiary on Instagram. Instead of thanking me, follow me!😅

**Reply by sab1a** [reply]
I see, for Algos, I've been reviewing DSA and hoping this would make the course during the sem more managable. CVIP is under Junsong Yuan. For the swap with an easier subject - is there any that you would recommend? I'm planning to work towards the AI Specialization. 

On browsing the course review, _Reinforcement Learning by Alina_ and _Modern Network Concepts by Lu Su_ were marked as easy but would like to know if any core software courses were classified as easy too? 

I'm considering swapping either _Introduction to Computer Vision and Image Processing_ or  _Distributed Systems_ with an easier subject.

Re Instagram: Followed you! Thanks for all you do! 😄

**Reply by snigi-gupta** [reply]
@sab1a ,
It is hard to say which course to take that is easy. Something course is easy but it has many deliverables, so that adds as an overhead. Best bet is you audit the course.

RL is really good specially coz of the professor. I would recommend it too. 

MNC is hard is what I have heard. I did not take either of these courses, so I am not 100% sure.

Also, is there a specific reason why you want to do AI/ML specialization? If the reason is getting an AI/ML job, then that is not possible that easily. I have rarely seen someone with ML/AI specialized MS get a job in AI/ML.
If you are not passionate about this field and won't pursue phd then AI/ML job is not for you. Just suggesting you to think about it. Most jobs in the US are for SDE and you'd want to target that.

**Reply by sab1a** [reply]
Sure, I will try to audit some courses during the first week. 

Re: the AI/ ML specialization, I liked the courses that were being offered under this specialization and since we had to choose base and advanced course, and I was more interested in what was being offered under AI/ML, I thought of choosing this as my specialization 

I reason I was not very keen on choosing software specialization was because I had already done some of the courses that were being offered under this in my Bachelor's degree - such as IR, OS. I also felt that with software development, I could take part in hackathons / competitions to demonstrate experience with specific web technologies (as I did in my Bachelors)

Although I will be mainly looking for roles as an SDE intern, I thought having the extra knowledge with AI/ML would expand the roles I apply to - eg. Research intern roles at companies. If I choose a SWE specialization, a lot of these RI roles require some AI/ML knowledge. 


But I'm still open to your thoughts and suggestions on what you think on the above thought process.
